lvjit/LuaFunctionBuilder.cpp

```diff
@@ -1633,6 +1689,115 @@ bool Lua::FunctionBuilder::do_forprep(TR::BytecodeBuilder* builder, Instruction
    return true;
 }
 
+bool Lua::FunctionBuilder::do_vararg(TR::BytecodeBuilder* builder, Instruction instruction) {
+   //int n = cast_int(base - ci->func) - cl->p->numparams - 1;
+   builder->Store("n",
+   builder->	Sub(
+   builder->		ConvertTo(Int32,
+   builder->                  Div(
+   builder->                      Sub(
+   builder->                          ConvertTo(Int64,
+   builder->                                    Load("base")),
+   builder->                          ConvertTo(Int64,
+   builder->                                    LoadIndirect("CallInfo", "func",
+   builder->                                                 Load("ci")))),
+   builder->                      ConstInt64(sizeof(TValue)))),
+   builder->        ConstInt32(prototype->numparams + 1)));
+
+   TR::IlBuilder *lessthan = nullptr;
+   builder->IfThen(&lessthan,
+   builder->       LessThan(
+   builder->                Load("n"),
+   builder->                ConstInt32(0)));
+
+   lessthan->Store("n",
+   lessthan->      ConstInt32(0));
+
+   int b = GETARG_B(instruction) - 1;  /* required results */
+   //if (b < 0) {  this is a compile time constant decision
+   if (b < 0) {
+      //b = n;  /* get all var. arguments */
+      builder->Store("b",
+      builder->      Load("n"));
+
+      //Protect(luaD_checkstack(L, n));
+      builder->Call("jit_checkstack", 2,
+      builder->     Load("L"),
+      builder->     Load("n"));
+      jit_Protect(builder);
+
+      //ra = RA(i);  /* previous call may change the stack */
+      builder->Store("ra", jit_R(builder, GETARG_A(instruction)));
+      //L->top = ra + n;
+      builder->StoreIndirect("lua_State", "top",
+      builder->              Load("L"),
+	  builder->              IndexAt(typeDictionary()->PointerTo(luaTypes.TValue),
+      builder->                      Load("base"),
+      builder->                      Add(
+      builder->                          ConstInt32(GETARG_A(instruction)),
+      builder->                          Load("n"))));
```

Comment:
Good call.  I copied the code from below.  Your code is clearer so I will change it.